internal/storage/v2/cassandra/factory.go

```diff
@@ -27,10 +28,10 @@ type Factory struct {
 }
 
 // NewFactory creates and initializes the factory
-func NewFactory(opts cassandra.Options, metricsFactory metrics.Factory, logger *zap.Logger) (*Factory, error) {
+func NewFactory(opts cassandra.Options, telset telemetry.Settings) (*Factory, error) {
 	f := &Factory{
-		metricsFactory: metricsFactory,
-		logger:         logger,
+		metricsFactory: telset.Metrics,
+		logger:         telset.Logger,
 	}
 	baseFactory, err := newFactoryWithConfig(opts, f.metricsFactory, f.logger)
```

Comment:
so the main thing I wanted to achieve is to pass the tracer to newFactoryWithConfig instead having it access a global variable. Then v1 factory wouldn't need to expose GetTracer function

Reply:
Yes. There's no longer a common Initializable interface.